### DISCUSSION TITLE
toggle_quick_terminal not working on gnome wayland

### DISCUSSION THREAD

**Discussion by clvx** [ghostty-org/ghostty#3459] [Q&A] [answered]
I have this setup in my config: `keybind = global:f12=toggle_quick_terminal` using gnome wayland; however, this is not working as desired.

I wonder if it's possible to trigger it using a gnome keyboard shortcut `Settings > Keyboard > Custom Shortcuts`. For now I have F12 mapped to ghostty but that opens a new terminal. I would like to map it to something like `ghostty toggle_quick_terminal`. 
If there's a cleaner way, I would love to be pointed out.

**Comment by gpanders** [comment]
The quick terminal is only supported on macOS right now.

**Reply by david2asher** [reply]
even on mac it has issues, after a while you are not focusing on the terminal, the hotkey for toggle_quick_terminal doesent work at all, the only way to make it work again is to manually focus on it again

**Comment by anon-legion** [comment]
Dang, had the same problem, not having this feature is a deal breaker for me. I'll wait for this feature to roll out on Linux before I start using Ghostty

**Reply by marcelhekking** [reply]
Mmm, same here :-(

**Comment by hedgieinsocks** [comment]
I can suggest that you try my gnome extension https://extensions.gnome.org/extension/6063/toggler/

I wrote it for myself, that's why it's dead simple. But perhaps it will be enough for you as well.

**Reply by prakash2033** [reply]
Great work, it works out of box. Where can we find all key combinations for your extension that is supported to use? 
I usually prefer `§` over `Ctrl+space` that you have as default

**Reply by hedgieinsocks** [reply]
@prakash2033 I presume you mean the backtick key typically located over the tab key. If so, then it is called `grave`. And you can set such single-key shortcut by simply typing the word `grave` there. Other special keys are `<Control>`, `<Shift>`, `<Alt>` and `<Super>`. You try to can combine them as you wish, e.g. `<Control><Shift>T` or `<Alt>grave`.

**Reply by hedgieinsocks** [reply]
There is also a similar project https://github.com/diegodario88/quake-terminal that goes way beyond my humble creation ;D

**Reply by prakash2033** [reply]
thank you for your prompt response. Unfortunately, it's not the grave key that I would like to use. I have a Nordic keyboard, and that has this key above tab § aka [section sign](https://en.m.wikipedia.org/wiki/Section_sign)

Edit: To reproduce my setup, you can add additional keyboard layout (swedish) in this case if that helps in helping me 😉

**Reply by hedgieinsocks** [reply]
Ah, `<Control>section` or just `section` is my first guess to try.

**Reply by prakash2033** [reply]
i will try this tomorrow and let you know if that works out,  thanks again.

**Reply by prakash2033** [reply]
`<ctrl>section` works perfectly. I also tried just `section`, which displays the terminal but doesn't hide it.

**Comment by david2asher** [comment]
even on mac it has issues, after a while you are not focusing on the terminal, the hotkey for toggle_quick_terminal doesent work at all, the only way to make it work again is to manually focus on it again

**Reply by pluiedev** [reply]
Please open a new discussion.

**Comment by pluiedev** [comment, marked as answer]
For future reference: while Ghostty 1.2 did add quick terminal support on Linux, GNOME Wayland is still not supported. This is because GNOME refuses to implement the [requisite Wayland protocol](https://wayland.app/protocols/wlr-layer-shell-unstable-v1) that allows applications to open windows on top of all existing windows. This is not the case on the vast majority of other Wayland-based desktop environments and compositors (e.g. KDE Plasma, Sway, Hyprland, Niri, etc.) and is therefore considered an upstream limitation that is beyond the scope of Ghostty. A GNOME Shell extension would likely be needed to implement this feature, but as of now we're not interested in maintaining one.
